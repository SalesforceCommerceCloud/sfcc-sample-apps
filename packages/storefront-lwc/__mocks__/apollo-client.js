import { register, ValueChangedEvent } from '@lwc/wire-service';

let mockedQueryResp = null;
let lastConnectedQueryConfig = null;

let mockedMutationResp = null;
let lastMutation = null;

export const useQuery = Symbol('apollo-use-query');
export const useMutation = Symbol('apollo-use-mutation');

export function reset() {
    mockedQueryResp = null;
    lastConnectedQueryConfig = null;

    mockedMutationResp = null;
    lastMutation = null;
}

export function mockMutation(resp) {
    mockedMutationResp = resp;
}

export function mockQuery(resp = {}) {
    mockedQueryResp = resp;
}

export function getLastMutation() {
    return lastMutation;
}

export function getLastConnectedConfig() {
    return lastConnectedQueryConfig;
}

register(useQuery, eventTarget => {
    function handleConnect() {
        if (mockedQueryResp) {
            eventTarget.dispatchEvent(
                new ValueChangedEvent({
                    client: {},
                    loading: false,
                    data: mockedQueryResp.data
                        ? mockedQueryResp.data
                        : mockedQueryResp,
                    error: mockedQueryResp.errors
                        ? mockedQueryResp.errors
                        : null,
                    initialized: true,
                    fetch: () => {},
                }),
            );
        }
    }
    function handleConfig(config) {
        lastConnectedQueryConfig = config;
    }
    function handleDisconnect() {}

    eventTarget.addEventListener('config', handleConfig);
    eventTarget.addEventListener('connect', handleConnect);
    eventTarget.addEventListener('disconnect', handleDisconnect);
});

register(useMutation, eventTarget => {
    function handleConnect() {
        if (mockedMutationResp) {
            eventTarget.dispatchEvent(
                new ValueChangedEvent({
                    client: {},
                    loading: false,
                    data: mockedMutationResp.data
                        ? mockedMutationResp.data
                        : mockedMutationResp,
                    error: mockedMutationResp.errors
                        ? mockedMutationResp.errors
                        : null,
                    initialized: true,
                    mutate: mutation => {
                        lastMutation = mutation;
                        return Promise.resolve();
                    },
                }),
            );
        }
    }
    function handleConfig(config) {}
    function handleDisconnect() {}

    eventTarget.addEventListener('config', handleConfig);
    eventTarget.addEventListener('connect', handleConnect);
    eventTarget.addEventListener('disconnect', handleDisconnect);
});
