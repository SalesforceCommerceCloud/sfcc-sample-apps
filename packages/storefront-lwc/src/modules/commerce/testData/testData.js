export const emptyBasketResponse = {
    errors: [
        {
            message: 'No Basket has been created yet.',
            locations: [{ line: 2, column: 3 }],
            path: ['getBasket'],
            extensions: {
                code: 'INTERNAL_SERVER_ERROR',
                exception: {
                    stacktrace: ['Error: No Basket has been created yet.'],
                },
            },
        },
    ],
    data: null,
};
export const basketWithItem = {
    data: {
        getBasket: {
            basketId: 'cb60c056280a77d090d5a25c59',
            customerId: 'bc5tfb0zfyMRyKrkXaxyha1Q3U',
            getBasketMessage: '',
            totalProductsQuantity: 1,
            shipmentId: 'me',
            shipmentTotal: 143.84,
            selectedShippingMethodId: '001',
            products: [
                {
                    productId: '701644031213M',
                    itemId: '7f3b564f59e7cea7462a9e30a5',
                    quantity: 1,
                    productName: 'Floral Dress',
                    price: 129,
                    imageURL: 'https://PZ.jpg',
                    inventory: {
                        ats: 100,
                        backorderable: false,
                        id: 'inventory_m',
                        orderable: true,
                        preorderable: false,
                        stockLevel: 100,
                        __typename: 'Inventory',
                    },
                    itemTotalAfterDiscount: 129,
                    itemTotalNonAdjusted: 129,
                    variationAttributes: [
                        {
                            id: 'color',
                            name: 'Color',
                            selectedValue: {
                                name: 'Hot Pink Combo',
                                orderable: true,
                                value: 'JJB52A0',
                                __typename: 'SelectedVariationValue',
                            },
                            __typename: 'ProductVariationAttribute',
                        },
                        {
                            id: 'size',
                            name: 'Size',
                            selectedValue: {
                                name: '6',
                                orderable: true,
                                value: '006',
                                __typename: 'SelectedVariationValue',
                            },
                            __typename: 'ProductVariationAttribute',
                        },
                    ],
                    prices: { list: null, sale: 129, __typename: 'Prices' },
                    productPromotions: null,
                    __typename: 'ProductItem',
                },
            ],
            orderTotal: 143.84,
            orderLevelPriceAdjustment: {
                itemText: '',
                price: 0,
                __typename: 'PriceAdjustment',
            },
            shippingLevelPriceAdjustment: {
                itemText: '',
                price: 0,
                __typename: 'PriceAdjustment',
            },
            shippingTotal: 7.99,
            shippingTotalTax: 0.4,
            taxation: null,
            taxTotal: 6.85,
            shippingMethods: {
                defaultShippingMethodId: '001',
                applicableShippingMethods: [
                    {
                        id: '001',
                        name: 'Ground',
                        description: 'Order received within 7-10 business days',
                        price: 7.99,
                        c_estimatedArrivalTime: '7-10 Business Days',
                        c_storePickupEnabled: null,
                        __typename: 'ShippingMethod',
                    },
                    {
                        id: '002',
                        name: '2-Day Express',
                        description: 'Order received in 2 business days',
                        price: 11.99,
                        c_estimatedArrivalTime: '2 Business Days',
                        c_storePickupEnabled: null,
                        __typename: 'ShippingMethod',
                    },
                    {
                        id: '003',
                        name: 'Overnight',
                        description: 'Order received the next business day',
                        price: 19.99,
                        c_estimatedArrivalTime: 'Next Day',
                        c_storePickupEnabled: null,
                        __typename: 'ShippingMethod',
                    },
                    {
                        id: '005',
                        name: 'Store Pickup',
                        description: 'Store Pickup',
                        price: 0,
                        c_estimatedArrivalTime: null,
                        c_storePickupEnabled: true,
                        __typename: 'ShippingMethod',
                    },
                    {
                        id: '012',
                        name: 'Express',
                        description:
                            'Orders shipped outside continental US received in 2-3 business days',
                        price: 22.99,
                        c_estimatedArrivalTime: '2-3 Business Days',
                        c_storePickupEnabled: null,
                        __typename: 'ShippingMethod',
                    },
                    {
                        id: '021',
                        name: 'USPS',
                        description:
                            'Order shipped by USPS received within 7-10 business days',
                        price: 7.99,
                        c_estimatedArrivalTime: '7-10 Business Days',
                        c_storePickupEnabled: null,
                        __typename: 'ShippingMethod',
                    },
                ],
                __typename: 'ShippingMethods',
            },
            couponItems: [],
            __typename: 'Basket',
        },
    },
};
