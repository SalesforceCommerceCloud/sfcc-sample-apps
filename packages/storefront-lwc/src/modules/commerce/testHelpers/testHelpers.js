export function tick(time = null) {
    return new Promise(resolve => setTimeout(resolve, time));
}
