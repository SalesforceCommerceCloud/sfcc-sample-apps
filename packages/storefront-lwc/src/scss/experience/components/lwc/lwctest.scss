.lwctest {
    color: chocolate;
    background-color: azure;
}
