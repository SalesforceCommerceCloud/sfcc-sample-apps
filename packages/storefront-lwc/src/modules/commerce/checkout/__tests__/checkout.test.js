import { createElement } from 'lwc';
import { mockQuery, reset } from '@lwce/apollo-client';
import Checkout from 'commerce/checkout';

import { registerWireService } from '@lwc/wire-service';
import { register } from 'lwc';
import { basketWithItem } from 'commerce/testData';
import { tick } from 'commerce/testHelpers';

registerWireService(register);

describe('<commerce-checkout>', function() {
    afterEach(() => {
        // The jsdom instance is shared across test cases in a single file so reset the DOM
        while (document.body.firstChild) {
            document.body.removeChild(document.body.firstChild);
        }

        reset();
    });

    it('should render a loading page', async function() {
        const element = createElement('commerce-checkout', {
            is: Checkout,
        });

        document.body.appendChild(element);

        await tick();

        expect(element.shadowRoot.querySelector('.loading')).toMatchSnapshot();
    });

    it('should initially show the shipping stage', async function() {
        mockQuery(basketWithItem);

        const element = createElement('commerce-checkout', {
            is: Checkout,
        });

        document.body.appendChild(element);

        await tick();

        expect(
            element.shadowRoot.querySelector('commerce-checkout-shipping'),
        ).toMatchSnapshot();
    });
});
