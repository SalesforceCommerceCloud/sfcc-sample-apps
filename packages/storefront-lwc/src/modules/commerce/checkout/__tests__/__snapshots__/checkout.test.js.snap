// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<commerce-checkout> should initially show the shipping stage 1`] = `
<commerce-checkout-shipping>
  #shadow-root(open)
    <div
      class="card"
    >
      Shipping 
    </div>
</commerce-checkout-shipping>
`;

exports[`<commerce-checkout> should render a loading page 1`] = `
<p
  class="text-center loading"
>
  Loading Checkout...
</p>
`;
